webpack/scenes/ContentViews/Details/Versions/Delete/RemoveSteps/CVEnvironmentSelectionForm.js

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useState, useContext } from 'react';
+import { Alert, Checkbox, EmptyState, EmptyStateVariant, Title, EmptyStateBody } from '@patternfly/react-core';
+import { TableVariant, TableComposable, Thead, Tbody, Tr, Th, Td } from '@patternfly/react-table';
+import { translate as __ } from 'foremanReact/common/I18n';
+import DeleteContext from '../DeleteContext';
+
+const CVEnvironmentSelectionForm = () => {
+  const {
+    versionNameToRemove, versionEnvironments, selected, setSelected,
+    setAffectedActivationKeys, setAffectedHosts, deleteFlow, setDeleteFlow,
+  } = useContext(DeleteContext);
+
+  const [allRowsSelected, setAllRowsSelected] = useState(false);
+
+  const onSelect = (event, isSelected, rowId) => {
+    const newSelected = selected.map((sel, index) => (index === rowId ? isSelected : sel));
+    setSelected(newSelected);
```

Comment:
If you use my `useSet` hook this could just be
```js
if (isSelected) {
  selected.add(rowId)
} else {
  selected.delete(rowId)
}
```
Just saying. :angel: